app/adapters/three_scale/api/instrumented_http_client.rb

```diff
@@ -58,13 +58,12 @@ def parse(response)
 
   protected
 
-  def zync_token
-    authentication = Rails.application.secrets.authentication || {}.freeze
-    authentication.fetch(:token) { return }
-  end
 
   def headers
-    super.merge('X-Request-Id' => SecureRandom.uuid, 'X-Zync-Token' => zync_token)
+    super.merge(
+      'X-Request-Id' => SecureRandom.uuid,
+      'X-Zync-Token' => Rails.application.secrets.authentication.fetch(:token) { return }
```

Comment:
This is going to raise a "NoMethodError: 'strip' is not defined for nil" when ENV['ZYNC_AUTHENTICATION_TOKEN'] is not defined.